src/system.js

```diff
@@ -65,80 +20,30 @@ const defaultConfig = {
-          getSafe(() => resolvedConfig.node.create.path),
-        selectors: {
-          title: 
-            process.env.TESTCAFE_DRUPAL_NODE_CREATE_SELECTORS_TITLE ||
-            getSafe(() => resolvedConfig.node.create.selectors.title),
-          save_button: 
-            process.env.TESTCAFE_DRUPAL_NODE_CREATE_SELECTORS_SAVE_BUTTON ||
-            getSafe(() => resolvedConfig.node.create.selectors.save_button)
-        }
-      }
-    },
-    users: {
-      admin: {
-        username:
-          process.env.TESTCAFE_DRUPAL_USERS_ADMIN_USERNAME ||
-          getSafe(() => resolvedConfig.users.admin.username),
-        password:
-          process.env.TESTCAFE_DRUPAL_USERS_ADMIN_PASSWORD ||
-          getSafe(() => resolvedConfig.users.admin.password),
-        role:
-          process.env.TESTCAFE_DRUPAL_USERS_ADMIN_ROLE ||
-          getSafe(() => resolvedConfig.users.admin.role)
-      },
-      editor: {
-        username:
-          process.env.TESTCAFE_DRUPAL_USERS_EDITOR_USERNAME ||
-          getSafe(() => resolvedConfig.users.editor.username),
-        password:
-          process.env.TESTCAFE_DRUPAL_USERS_EDITOR_PASSWORD ||
-          getSafe(() => resolvedConfig.users.editor.password),
-        role:
-          process.env.TESTCAFE_DRUPAL_USERS_EDITOR_ROLE ||
-          getSafe(() => resolvedConfig.users.editor.role)
-      },
-      authenticated_user: {
-        username:
-          process.env.TESTCAFE_DRUPAL_USERS_AUTHUSER_USERNAME ||
-          getSafe(() => resolvedConfig.users.authenticated_user.username),
-        password:
-          process.env.TESTCAFE_DRUPAL_USERS_AUTHUSER_PASSWORD ||
-          getSafe(() => resolvedConfig.users.authenticated_user.password)
-      }
-    },
-    user: {
-      login: {
-        path:
-          process.env.TESTCAFE_DRUPAL_USER_LOGIN_PATH ||
-          getSafe(() => resolvedConfig.user.login.path),
-        selectors: {
-          username:
-            process.env.TESTCAFE_DRUPAL_USER_LOGIN_SELECTORS_USERNAME ||
-            getSafe(() => resolvedConfig.user.login.selectors.username),
-          password:
-            process.env.TESTCAFE_DRUPAL_USER_LOGIN_SELECTORS_PASSWORD ||
-            getSafe(() => resolvedConfig.user.login.selectors.password),
-          login_button:
-            process.env.TESTCAFE_DRUPAL_USER_LOGIN_SELECTORS_LOGINBUTTON ||
-            getSafe(() => resolvedConfig.user.login.selectors.login_button)
-        }
-      },
-      add: {
-        path:
-          process.env.TESTCAFE_DRUPAL_USER_ADD_PATH ||
-          getSafe(() => resolvedConfig.user.add.path)
-      }
-    }
-  };
+  // Allow config overrides using environmental variables.
+  if (typeof process.env.TESTCAFE_BASEURL !== "undefined") {
+    resolvedConfig.baseUrl = process.env.TESTCAFE_BASEURL;
+  }
+  if (typeof process.env.TESTCAFE_DRUPAL_USERS_ADMIN_USERNAME !== "undefined") {
+    resolvedConfig.users.admin.username = process.env.TESTCAFE_DRUPAL_USERS_ADMIN_USERNAME;
```

Comment:
My leaning generally for all JS stuff is to follow a functional pattern where it's reasonable, I'm not suggesting that we change this code here, but do need to keep in mind that if we ever read back `defaultConfig` the value might be mutated due to the shallow copy nature of `Object.assign`

<img width="539" alt="Screen Shot 2019-10-01 at 11 08 51 am" src="https://user-images.githubusercontent.com/14073294/65931362-f5d39b80-e43b-11e9-8095-93a58946d5f5.png">

Reply:
@yjimk I have gone ahead with the current changes, but am curious to understand your comment here about shallow copy nature of Object.assign(). Perhaps one day when you have time, you can educate me on it :-)